bokehjs/test/unit/models/tools/inspectors/hover_tool.ts

```diff
@@ -39,21 +38,21 @@ describe("HoverTool", () => {
     it("should invalidate tooltips' template when changing the tooltips property", async () => {
       const {hover_view, data_source} = await make_testcase()
 
-      const el0 = hover_view._render_tooltips(data_source, 0, {index: 0, x: 123, y: 456, sx: 0, sy: 0})
+      const el0 = hover_view._render_tooltips(data_source, ({index: 0, x: 123, y: 456, sx: 0, sy: 0} as TooltipVars))
```

Comment:
Parentheses are redundant.

Reply:
Instead of casting, required fields should be filled in or alternatively marked as optional (maybe there are too many non-optional fields there). Otherwise, this test makes too many assumptions about the implementation, giving false sense of security on the type-level.

Reply:
Definitely not too many non-optional fields. The main point of these changes is to make the set of tooltip vars as uniform as possible across all glyphs.